### ISSUE TITLE
PVC for projects directory with Ephemeral=false is not owned by the Deployment

### ISSUE THREAD

**Issue by feloy** [redhat-developer/odo#5886]
/kind bug

## How did you run odo exactly?

```
$  odo preference set Ephemeral false

$ odo dev
```

```
kubectl get pvc odo-projects-nodejs-ex-app -o yaml
```

## Actual behavior

The pvc does not contain any OwnerRefernce

## Expected behavior

The pvc should contain an OwnerReference

**Comment by rm3l** [comment]
/priority High

As discussed, this can be problematic because `Ephemeral` is now set to `false` by default (as of #5795 - #5241)

**Comment by feloy** [comment]
To lativize, that happens only when the Name into the env.yaml file is different from the Devfile metadata.name. It should be resolved with https://github.com/redhat-developer/odo/pull/5852

**Comment by rm3l** [comment]
> To lativize, that happens only when the Name into the env.yaml file is different from the Devfile metadata.name. It should be resolved with #5852

Thanks for the information. I'll link ##5852 to this issue as well then, so it can be closed once done.

**Comment by valaparthvi** [comment]
I can confirm that this issue has been fixed with #5852
